File: tyama_icpc2005dA(PKU2683).pas
(*2005 Domestic A Žš‰º‚°‚ÍHaskell•—‚Éw*)
{$apptype console} //Delphi‚Ì‚¨‚Ü‚¶‚È‚¢

program DomesticA;
var n,i: integer;

function parse(m,y:longint):longint;
var
t,c,i,dm:longint;
r: real;
begin
dm:=0;
readln(t,r,c);
if(t<>0) then begin
  for i:=1 to y do begin
    dm:=trunc(r*m);
    m:=m+dm-c;
    end;
  end else begin
  for i:=1 to y do begin
    dm:=dm+trunc(r*m);
    m:=m-c;
    end;
  m:=m+dm;
  end;
parse:=m;
end;

procedure data();
var m,y,n,i,r,x: longint;
begin
read(m,y,n);
r:=m;
for i:=1 to n do begin
  x:=parse(m,y);
  if(x>r)then r:=x;
  end;
writeln(r);
end;

begin
read(n);
for i:=1 to n do data();
end.

File: tyama_icpc2005dA(PKU2683).java
//2005 Domestic A
import java.io.*;
import java.util.*;
class Main{
  public static Scanner cin;
  public static int parse(int m, int y){
    int t,c,dm=0;
    double r;
    t=cin.nextInt();r=cin.nextDouble();c=cin.nextInt();
    if(t!=0){
      for(;y!=0;y--){
        dm=(int)(r*m);
        m+=dm-c;
      }
    }else{
       for(;y!=0;y--){
         dm=dm+(int)(r*m);
         m-=c;
       }
       m+=dm;
    }
    return m;
}

  public static void data(){
    int m,y,n,r,x;
    r=m=cin.nextInt();y=cin.nextInt();n=cin.nextInt();
    for(;n!=0;n--){
      x=parse(m,y);
      if(x>r)r=x;
    }
    System.out.println(r);
  }

  public static void main(String args[]){
    cin=new Scanner(System.in);
    int n=cin.nextInt();
    for(;n!=0;n--)data();
  }
}